src/inspector_agent.cc

```diff
@@ -648,6 +659,101 @@ void Agent::AllAsyncTasksCanceled() {
   client_->AllAsyncTasksCanceled();
 }
 
+void Agent::InterceptAsyncStackDepthMessage(const StringView& message) {
+  // This logic would be better implemented in JavaScript, but when using
+  // --inspect-brk, the debugger must necessarily attach before much JavaScript
+  // can execute. The Debugger.setAsyncCallStackDepth message arrives too early
+  // and we must intercept this in C++.
+
+  v8::Isolate* isolate = parent_env_->isolate();
+  Local<Context> context = parent_env_->context();
+
+  MaybeLocal<String> string =
+    String::NewFromTwoByte(isolate, message.characters16(),
+                           v8::NewStringType::kNormal, message.length());
+  if (string.IsEmpty()) {
+    return;
+  }
+
+  // Basically, the logic we want to implement is:
+  // let parsed; try {
+  //   parsed = JSON.parse(string);
+  // } catch (e) { return; }
+  // if (parsed && parsed.method && parsed.method === 'Debugger.setAsync..' &&
+  //     parsed.params && parsed.params.maxDepth &&
+  //     typeof parsed.params.maxDepth === 'number') {
+  //   // Enable or Disable, depending on maxDepth.
+  // }
+  //
+  // We ignore (return early) on malformed messages and let v8-inspector deal
+  // with them.
+
+  MaybeLocal<Value> maybe_parsed =
+    v8::JSON::Parse(context, string.ToLocalChecked());
+  if (maybe_parsed.IsEmpty()) {
+    return;
+  }
+
+  Local<Value> parsed = maybe_parsed.ToLocalChecked();
+  if (!parsed->IsObject()) {
+    return;
+  }
+
+  Local<Object> object = parsed.As<Object>();
+
+  Local<Value> method;
+  if (!object->Get(context, parent_env_->method_string()).ToLocal(&method) ||
+      !method->IsString() ||
+      !method->StrictEquals(parent_env_->async_stack_depth_string())) {
+    return;
+  }
+
+  Local<Value> params;
+  if (!object->Get(context, parent_env_->params_string()).ToLocal(&params) ||
+      !params->IsObject()) {
+    return;
+  }
+
+  Local<Value> depth_value;
+  if (!params.As<Object>()->Get(context, parent_env_->max_depth_string())
+                           .ToLocal(&depth_value) ||
+      !depth_value->IsNumber()) {
+    return;
+  }
+
+  Maybe<double> maybe_depth = depth_value->NumberValue(context);
+  if (maybe_depth.IsNothing()) {
+    return;
```

Comment:
I think this can’t actually be the case since you checked `depths_value->IsNumber()`

Reply:
Done.